### ISSUE TITLE
Support `when` in ConfigOptions `groups`

### ISSUE THREAD

**Issue by aj-jester** [replicatedhq/kots#725]
Currently the `groups` in ConfigOptions does not seem to support `when` and this seems to be a pretty common request.

https://github.com/replicatedhq/kots/blob/master/kotskinds/apis/kots/v1beta1/config_types.go#L59

```
spec:
  groups:
   - name: advanced_configuration
      title: Advanced Configuration
      when:  'repl{{ ConfigOptionEquals "advanced_configuration" "1"}}' <<<<----- Need to support this.
      items:
        - name: manager_cpu
          title: Manager CPU
          type: text
          default: "2000"
        - name: verification_cpu
          title: Verification Service CPU
          type: text
          default: "2000"
```

**Comment by aj-jester** [comment]
Possible PR https://github.com/replicatedhq/kots/pull/326

**Comment by marccampbell** [comment]
Fixed in 1.17
